cpp/src/arrow/ipc/reader.cc

```diff
@@ -290,6 +296,7 @@ class ArrayLoader {
 
   template <typename TYPE>
   Status LoadBinary(Type::type type_id) {
+    DCHECK_NE(nullptr, out_);
```

Comment:
Minor nit: would you putting the value checked on the left hand side?
```suggestion
    DCHECK_NE(out_, nullptr);
```

Reply:
done!

Reply:
Sorry, I've update it now